org.eclipse.jdt.apt.core/src/org/eclipse/jdt/apt/core/util/AptConfig.java

```diff
@@ -364,8 +364,8 @@ private static void addProjectClasspath(
     		IPath binPath = otherJavaProject.getOutputLocation();
     		IResource binPathResource = root.findMember(binPath);
     		String binDirString;
-    		if (binPathResource != null) {
-    			binDirString = root.findMember(binPath).getLocation().toOSString();
+    		if (binPathResource != null && binPathResource.getLocation() != null) {
```

Comment:
The fix is formally correct.
However, the `else` branch will still use same (not existing) path.
Why should the `else` branch work? Because someone creates the path **later on**?

Reply:
Why is the path in the else branch not existing?
The NPE was in the if branch which means that the binPath exists and also a member was found (binPathResource).

Or do you mean that if the binPathResource is found but the location can not be determined, the method should be aborted (by return)?

Reply:
`binPath` is just a string value that we are trying to resolve via `root.findMember(binPath).getLocation()`.

So if there are no path variables used in the location, it is **same** path.

So if the `binPathResource.getLocation()` can't be resolved, it means it is either doesn't exist (anymore?) or path variable is wrong, or it is not local one.

So the "else" branch **seem** to be wrong in all of such cases.

=> therefore I wonder if you can provide the case where it would work.